File: flutter/lib/core/benchmark/log_exec_duration.dart
import 'package:flashcard/core/config/di/config_di.dart';
import 'package:flutter/foundation.dart';
import 'package:logger/logger.dart';

Future<T> logExecDuration<T>(
    Future<T> Function() action, {
      String name = 'no_name',
      String tag = 'no_tag',
      bool log = true,
      Logger? logger,
    }) async {
  final stopwatch = Stopwatch()..start();
  final result = await action();
  stopwatch.stop();

  logger ??= getIt<Logger>();

  if (log && kDebugMode) {
    logger.d('$tag | Execution time for $name: ${stopwatch.elapsedMilliseconds} ms');
  }
  return result;
}

File: react-native/src/core/benchmark/logExecDuration.ts
export async function logExecDuration<T>(
  action: () => Promise<T>,
  opts: { name?: string; tag?: string; log?: boolean } = {},
): Promise<T> {
  const { name = 'no_name', tag = 'no_tag', log = true } = opts;
  const start = Date.now();
  const result = await action();
  const elapsed = Date.now() - start;
  if (log) {
    // eslint-disable-next-line no-console
    console.log(`${tag} | Execution time for ${name}: ${elapsed} ms`);
  }
  return result;
}


